app/src/main/java/de/xikolo/App.kt

```diff
@@ -102,4 +116,37 @@ class App : Application() {
         CookieManager.getInstance().flush()
     }
 
+    @RequiresApi(Build.VERSION_CODES.N_MR1)
+    private fun configureShortcuts() {
+        val recentCoursesObserver = Observer<String> {
+            updateShortcuts()
+        }
+
+        val recentCoursesStorage = RecentCoursesStorage()
+        recentCoursesStorage.coursesLive.observeForever(recentCoursesObserver)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.N_MR1)
+    private fun updateShortcuts() {
+        val shortcutManager = getSystemService<ShortcutManager>(ShortcutManager::class.java)
+        val recentCourses = RecentCoursesStorage().recentCourses
+        val intentList = mutableListOf<ShortcutInfo>()
+
+        for (course in recentCourses) {
+            val courseId = course.first
+            val title = course.second
+
+            val intent = CourseActivityAutoBundle.builder().courseId(courseId).build(applicationContext)
+            intent.action = Intent.ACTION_VIEW
+
+            val shortcut = ShortcutInfo.Builder(applicationContext, courseId)
+                .setShortLabel(title)
```

Comment:
shortLabel: `A concise phrase that describes the shortcut's purpose. When possible, limit the length of the "short description" of a shortcut to 10 characters.`
Aren't our titles a bit longer than that?

Reply:
Yes, I see that problem too. The IOS screenshot in #209 also uses the full titles. Alternatively, we could use the label "Continue" but then we could only show the latest course. In my opinion it would be better to show the last three courses with a cut off title than only one. What do you think?

Reply:
[`ShortcutInfo.Builder`](https://developer.android.com/reference/android/content/pm/ShortcutInfo.Builder) also provides  `setLongLabel`. However, since we display up to 3 courses, we cannot omit the title.